sycl/source/detail/graph/graph_impl.cpp

```diff
@@ -442,7 +432,9 @@ graph_impl::add(std::function<void(handler &)> CGF,
 
   // Pass the node deps to the handler so they are available when processing the
   // CGF, need for async_malloc nodes.
-  Handler.impl->MNodeDeps = Deps;
+  Handler.impl->MNodeDeps.clear();
```

Comment:
This is to preserve the semantics. Not sure if truly necessary, maybe there is a guarantee that it must be empty. And if not, why are we dropping old deps?

Reply:
Good catch. My understanding is that this is only needed for `async_malloc` nodes, which implies only one instance of an executable graph is allowed.
`MNodeDeps` is empty at that point for all existing tests, so either we do have a guarantee or gap in our tests.
We might want to either preserve semantics or add an assertion ...

@Bensuo any thoughts?